Source/Internal/IGListPreprocessor.h

```diff
@@ -0,0 +1,34 @@
+//
+//  IGListPreprocessor.h
+//  IGListKit
+//
+//  Created by Adlai Holler on 2/20/17.
+//  Copyright © 2017 Instagram. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+#import <IGListKit/IGListMacros.h>
+#import <IGListKit/IGListPreprocessingContext.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class IGListSectionController, IGListSectionMap;
+
+IGLK_SUBCLASSING_RESTRICTED
+@interface IGListPreprocessor : NSObject
+
+- (instancetype)initWithSectionMap:(IGListSectionMap *)sectionMap;
+
+/**
+ * Must be called from the main thread.
+ */
+- (void)schedulePreprocessingWithContainerSize:(CGSize)containerSize completion:(dispatch_block_t)completionBlock;
```

Comment:
@amonshiz this is the only way to do it w/out making that a locking property. IMO this is much cleaner, but probably something we need to consider supporting down the line. This is following my outline in #383 btw.